templates/s3/s3.py

```diff
@@ -77,16 +76,16 @@ def _nodes(self):
 
     @property
     def _minio(self):
-        if self.__minio is None:
-            if self.data['minioLocation']['nodeId'] and self.data['minioLocation']['robotURL']:
-                try:
-                    robot = self.api.robots.get(
-                        self.data['minioLocation']['nodeId'],
-                        self.data['minioLocation']['robotURL'])
-                    self.__minio = robot.services.get(template_uid=MINIO_TEMPLATE_UID, name=self.guid)
-                except ConnectionError:
-                    self.state.set('status', 'running', 'error')
-        return self.__minio
+        if self.data['minioLocation']['nodeId'] and self.data['minioLocation']['robotURL']:
+            try:
+                robot = self.api.robots.get(
+                    self.data['minioLocation']['nodeId'],
+                    self.data['minioLocation']['robotURL'])
+                return robot.services.get(template_uid=MINIO_TEMPLATE_UID, name=self.guid)
+            except ConnectionError:
+                self.state.set('status', 'running', 'error')
+            except ServiceNotFoundError:
+                return self._deploy_minio()
```

Comment:
This means you're going to install a new minio possibly on a new node and not the same where we were expecting to find it. I wonder if this is ok :-/

Reply:
It shouldn't cause any issues. The connection info will be updated in the service data. `s3_redundant` queries `s3` for connection so there shouldn't be issues there. The only thing I can think of is that a user will need to query the s3 for the new url. Also, the url will need to be updated in the reverse_proxy, right?

Reply:
But updating the reverse proxy will be needed anyway even if it was created on the same node, since the url will change if it has a new port.